I tried to build `cppad:x86-windows`  on a fresh windows installation and I got the following error. @NancyLi1013 could you please give me an explanation of why this happening?

Should I remove `PREFER_NINJA` from `vcpkg_configure_cmake` when the triplet `x86-windows` is used?
```
.\vcpkg.exe install cppad:x86-windows
Computing installation plan...
The following packages will be built and installed:
    cppad[core]:x86-windows
Detecting compiler hash for triplet x86-windows...
Starting package 1/1: cppad:x86-windows
Building package cppad[core]:x86-windows...
Could not locate cached archive: C:\Users\gromualdi\AppData\Local\vcpkg\archives\60\60acc7c8c365356ea3fa728f338186ea7a673064.zip
-- Using cached C:/Users/gromualdi/robot-code/vcpkg/downloads/coin-or-CppAD-20200000.3.tar.gz
-- Cleaning sources at C:/Users/gromualdi/robot-code/vcpkg/buildtrees/cppad/src/20200000.3-2149299b1e.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source C:/Users/gromualdi/robot-code/vcpkg/downloads/coin-or-CppAD-20200000.3.tar.gz
-- Applying patch windows-fix.patch
-- Using source at C:/Users/gromualdi/robot-code/vcpkg/buildtrees/cppad/src/20200000.3-2149299b1e.clean
-- Configuring x86-windows
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:72 (message):
    Command failed: ninja -v
    Working Directory: C:/Users/gromualdi/robot-code/vcpkg/buildtrees/cppad/x86-windows-rel/vcpkg-parallel-configure
    Error code: 1
    See logs for more information:
      C:\Users\gromualdi\robot-code\vcpkg\buildtrees\cppad\config-x86-windows-out.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_configure_cmake.cmake:300 (vcpkg_execute_required_process)
  ports/cppad/portfile.cmake:13 (vcpkg_configure_cmake)
  scripts/ports.cmake:79 (include)
```

--- a/ports/cppad/CONTROL
+++ b/ports/cppad/CONTROL
@@ -2,3 +2,4 @@ Source: cppad
 Version: 20200000.3
 Homepage: https://github.com/coin-or/CppAD
 Description: CppAD: A Package for Differentiation of C++ Algorithms
+Supports: !(arm|uwp)

--- a/ports/cppad/pkgconfig-fix.patch
+++ b/ports/cppad/pkgconfig-fix.patch
@@ -1,0 +1,63 @@
+diff --git a/pkgconfig/CMakeLists.txt b/pkgconfig/CMakeLists.txt
+index db8b65902..b1224a01d 100644
+--- a/pkgconfig/CMakeLists.txt
++++ b/pkgconfig/CMakeLists.txt
+@@ -42,8 +42,9 @@ ENDIF( cppad_has_ipopt )
+ # cppad_pkgconfig_cflags
+ # cppad_pkgconfig_cflags_uninstalled
+ #
+-SET(cppad_pkgconfig_cflags             "-I${cppad_abs_includedir}" )
+-SET(cppad_pkgconfig_cflags_uninstalled "-I${cppad_SOURCE_DIR}"     )
++SET(cppad_pkgconfig_includedir          "${cmake_install_includedirs}" )
++SET(cppad_pkgconfig_cflags_uninstalled  "-I${cppad_SOURCE_DIR}"        )
++SET(cppad_pkgconfig_libdir              "${cmake_install_libdirs}"     )
+ # ----------------------------------------------------------------------------
+ # cppad_pkgconfig_libs
+ # cppad_pkgconfig_libs_uninstalled
+@@ -74,7 +75,7 @@ IF( "${libs}" STREQUAL "" )
+     SET(cppad_pkgconfig_libs             "")
+     SET(cppad_pkgconfig_libs_uninstalled "")
+ ELSE( "${libs}" STREQUAL "" )
+-    SET(cppad_pkgconfig_libs             "-L${cppad_abs_libdir} ${libs}")
++    SET(cppad_pkgconfig_libs             "-L\${libdir} ${libs}")
+     SET(cppad_pkgconfig_libs_uninstalled "${uninstalled} ${libs}")
+ ENDIF( "${libs}" STREQUAL "" )
+ #
+@@ -84,10 +85,12 @@ ENDIF( "${libs}" STREQUAL "" )
+ CONFIGURE_FILE(
+     ${CMAKE_CURRENT_SOURCE_DIR}/cppad.pc.in
+     ${CMAKE_CURRENT_BINARY_DIR}/cppad.pc
++    @ONLY
+ )
+ CONFIGURE_FILE(
+     ${CMAKE_CURRENT_SOURCE_DIR}/cppad-uninstalled.pc.in
+     ${CMAKE_CURRENT_BINARY_DIR}/cppad-uninstalled.pc
++    @ONLY
+ )
+ # During install copy cppad.pc to datadir and libdir
+ INSTALL(FILES ${CMAKE_CURRENT_BINARY_DIR}/cppad.pc
+diff --git a/pkgconfig/cppad.pc.in b/pkgconfig/cppad.pc.in
+index a202e3fb3..1c34c1e08 100644
+--- a/pkgconfig/cppad.pc.in
++++ b/pkgconfig/cppad.pc.in
+@@ -12,12 +12,19 @@
+ # This file is used with pkg-config to include CppAD after it is installed
+ #
+ # Entries set by top source directory CMakeLists.txt file
++
++prefix=${pcfiledir}/../..
++exec_prefix=${prefix}
++libdir=${exec_prefix}/@cppad_pkgconfig_libdir@
++includedir=${prefix}/@cppad_pkgconfig_includedir@
++
+ Name:         cppad
+ Description:  @cppad_description@
+ Version:      @cppad_version@
+ URL:          @cppad_url@
++
+ #
+ # Entries set by pkgconfig/CMakeLists.txt file
+-Cflags:       @cppad_pkgconfig_cflags@
++Cflags:       -I${includedir}
+ Libs:         @cppad_pkgconfig_libs@
+ Requires:     @cppad_pkgconfig_requires@

--- a/ports/cppad/portfile.cmake
+++ b/ports/cppad/portfile.cmake
@@ -6,6 +6,7 @@ vcpkg_from_github(
     HEAD_REF master
     PATCHES
         windows-fix.patch
+        pkgconfig-fix.patch
 )
 
 vcpkg_configure_cmake(

--- a/ports/cppad/portfile.cmake
+++ b/ports/cppad/portfile.cmake
@@ -16,6 +16,7 @@ vcpkg_from_github(
 vcpkg_configure_cmake(
     SOURCE_PATH ${SOURCE_PATH}
     PREFER_NINJA
+    DISABLE_PARALLEL_CONFIGURE
     OPTIONS
         -Dcppad_prefix=${CURRENT_PACKAGES_DIR}
     OPTIONS_RELEASE

--- a/ports/cppad/portfile.cmake
+++ b/ports/cppad/portfile.cmake
@@ -1,3 +1,7 @@
+# the compilation fails on arm and uwp. Please check the related issue:
+# https://github.com/microsoft/vcpkg/pull/12560#issuecomment-668412073
+vcpkg_fail_port_install(ON_TARGET "uwp" and "arm")
+
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO coin-or/CppAD

